include/nighthawk/common/milestone_tracker.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
```

Comment:
Yeah, I reckon it makes sense to split out the milestone tracker from the pool PR.